### PR TITLE
Migrate newsletter preference center to Fluent (Fixes #9401)

### LINKED ISSUE #9401: Requesting update to Internet Health Report newsletter title and description on email preference center
Opening issue per discussion with @pmac

Current copy for newsletter.
**Title:** Internet Health Report
**Description:** Keep up with our annual compilation of research and stories on the issues of privacy & security, openness, digital inclusion, decentralization, and web literacy. 

Updated copy for newsletter.
**Title:** Insights
**Description:** Mozilla publishes articles and deep dives on issues around internet health and trustworthy AI, including our annual Internet Health Report.

### PR DESCRIPTION
**Pull request by alexgibson**
## Description
- Migrates `/newsletter/` app from .lang to Fluent

## Issue / Bugzilla link
#8659
#9401

## Testing
- http://localhost:8000/en-US/newsletter/
- http://localhost:8000/en-US/newsletter/existing/*TOKEN*/
- http://localhost:8000/en-US/newsletter/confirm/*TOKEN*/
- http://localhost:8000/en-US/newsletter/country/*TOKEN*/
- http://localhost:8000/en-US/newsletter/updated/
- http://localhost:8000/en-US/newsletter/recovery/
- http://localhost:8000/en-US/newsletter/mozilla/
- http://localhost:8000/en-US/newsletter/firefox/
- http://localhost:8000/en-US/newsletter/developer/
- http://localhost:8000/en-US/newsletter/fxa-error/
- http://localhost:8000/en-US/newsletter/opt-out-confirmation/
- http://localhost:8000/en-US/newsletter/country/success/

### CODE REVIEW

File: bedrock/mozorg/forms.py [hunk 1]
```diff
@@ -54,7 +50,7 @@ class HoneyPotWidget(widgets.TextInput):
     """Render a text field to (hopefully) trick bots. Will be used on many pages."""
 
     def render(self, name, value, attrs=None, renderer=None):
-        honeypot_txt = _(u'Leave this field empty.')
+        honeypot_txt = 'Leave this field empty.'
```
Comment (alexgibson) [inline bedrock/mozorg/forms.py:53]: This string appears never to have been localized

File: bedrock/newsletter/forms.py [hunk 2]
```diff
@@ -74,8 +71,8 @@ class BooleanTabularRadioSelect(widgets.RadioSelect):
 
     def __init__(self, attrs=None):
         choices = (
-            ('true', _('Yes')),
-            ('false', _('No')),
+            ('true', 'Yes'),
```
Comment (alexgibson) [inline bedrock/newsletter/forms.py:74]: Strings never localized

File: bedrock/newsletter/forms.py [hunk 3]
```diff
@@ -74,8 +71,8 @@ class BooleanTabularRadioSelect(widgets.RadioSelect):
...
+            ('false', 'No'),
```
Comment (alexgibson) [inline bedrock/newsletter/forms.py:75]: Strings never localized

File: bedrock/newsletter/forms.py [hunk 4]
```diff
@@ -203,7 +200,7 @@ def clean(self):
         valid_newsletters = utils.get_newsletters()
         for newsletter in self.newsletters:
             if newsletter not in valid_newsletters:
-                msg = _("%s is not a valid newsletter") % newsletter
+                msg = ftl('newsletters-is-not-a-valid-newsletter', newsletter=newsletter)
```
Comment (alexgibson) [inline bedrock/newsletter/forms.py:203]: I'm not sure how to test this message?

File: bedrock/newsletter/templates/newsletter/updated.html [hunk 5]
```diff
@@ -58,56 +56,56 @@ <h4>{{_('Would you mind telling us why you’re leaving?') }}</h4>
...
+            <h1>{{ ftl('newsletters-youve-been-unsubscribed') }}</h1>
+            <h2>{{ ftl('newsletters-were-sorry-to-see-you-go') }}</h2>
+            <p>{{ ftl('newsletters-thanks-for-telling-us-why') }}</p>
           </header>
         </section>
       </div>
     {% else %}
       <div class="mzp-l-content">
         <section class="c-updated-block">
-          {% if l10n_has_tag('newsletter_confirm_0320') %}
```
Comment (alexgibson) [inline bedrock/newsletter/templates/newsletter/updated.html:70]: Looks like this conditional copy was never exposed for L10n, so changing it to hard-coded here for now.

File: bedrock/newsletter/tests/test_views.py [hunk 6]
```diff
@@ -513,10 +512,6 @@ def test_unknown_email(self, mock_basket):
         rsp = self.client.post(self.url, data)
         self.assertTrue(mock_basket.called)
         self.assertEqual(200, rsp.status_code)
-        form = rsp.context['form']
-        expected_error = unknown_address_text % \
-            reverse('newsletter.subscribe')
-        self.assertIn(expected_error, form.errors['email'])
```
Comment (alexgibson) [inline bedrock/newsletter/tests/test_views.py:519]: I wasn't sure how to test this with a Fluent string - is it still possible?

File: bedrock/newsletter/views.py [hunk 7]
```diff
@@ -362,11 +337,11 @@ def existing(request, token=None):
                 else:
                     title = nstrings['title']
                 description = nstrings.get('description', u'')
-            else:
+            # else:
                 # Firefox Marketplace for Desktop/Android/Firefox OS should be
                 # shorten in the titles
-                title = _(data['title'].replace('Firefox Marketplace for ', ''))
-                description = _(data['description'])
+                # title = _(data['title'].replace('Firefox Marketplace for ', ''))
```
Comment (alexgibson) [inline bedrock/newsletter/views.py:343]: I'm not sure how to handle this with Fluent - but I also thought it probably isn't needed today anyway? Thoughts?

File: bedrock/newsletter/views.py [hunk 8]
```diff
@@ -36,163 +35,140 @@
...
-# NOTE: Must format a link into this: (https://www.mozilla.org/newsletter/)
-unknown_address_text = _lazy(
-    u'This email address is not in our system. Please double check your '
-    u'address or <a href="%s">subscribe to our newsletters.</a>')
-
-invalid_email_address = _lazy(u'This is not a valid email address. '
-                              u'Please check the spelling.')
+FTL_FILES = ['mozorg/newsletters']
+
+general_error = ftl_lazy('newsletters-we-are-sorry-but-there', ftl_files=FTL_FILES)
```
Comment (alexgibson) [inline bedrock/newsletter/views.py:40]: Not sure if there is a cleaner way than adding `ftl_files=FTL_FILES` to each of these `ftl_lazy()` calls?

**Comment by alexgibson** [comment]
@pmac ok I've given this a decent amount of testing and things seems to be working ok. I've added the new strings requested in https://github.com/mozilla/bedrock/issues/9401 as well as a bunch of other strings that were never localized.

I haven't gone as far as tidying up the .ftl files yet and running the migration, as I figured it might be a good idea to get a review here first. Mostly I'm curious if a) I've missed/overlooked anything important, b) if I've broken something unsuspectingly.

File: bedrock/newsletter/views.py (hunk 8 above)
Comment (pmac) [reply bedrock/newsletter/views.py:40]: Unfortunately not at the moment. If `newsletters` is going to be a default file then we could omit the `ftl_files` param though.

File: bedrock/newsletter/views.py [hunk 9]
```diff
@@ -612,7 +590,9 @@ def recovery(request):
                     # Tell them, give them a link to go subscribe if they want
                     url = reverse('newsletter.subscribe')
                     form.errors['email'] = \
-                        form.error_class([unknown_address_text % url])
+                        form.error_class([ftl_lazy('newsletters-this-email-address-is-not',
```
Comment (pmac) [inline bedrock/newsletter/views.py:593]: no need to use `ftl_lazy` here since it's in the view.

File: bedrock/newsletter/views.py (hunk 7 above)
Comment (pmac) [reply bedrock/newsletter/views.py:349]: I think you're right. We should be fine to remove this.

**Review by pmac — APPROVED** [review]
This is looking great! Thanks for all this work! 👍

File: bedrock/newsletter/tests/test_views.py (hunk 6 above)
Comment (pmac) [reply bedrock/newsletter/tests/test_views.py:519]: I guess we could assert that the error text matches what we expect... or just call the `ftl()` here to get the text.

**Comment by alexgibson** [comment]
Thanks @pmac - I've run the migration and opened a PR against the L10n repo. Once Peiying has reviewed the strings, I think this is ready.

**Comment by alexgibson** [comment]
@pmac ok migration has been reviewed & merged, so I think this should be ready for a final look?

**Comment by pmac** [comment]
Everything looks good to me, and works well in my local testing. 🎃 :shipit:

**Pull request merged**